tests/integration/test_kms.py

```diff
@@ -220,3 +220,25 @@ def test_list_aliases_of_key(self, kms_client, kms_create_key):
 
         response = kms_client.list_aliases(KeyId=comparison_key["KeyId"])
         assert len(response["Aliases"]) == 0
+
+    # Key ARNs, key IDs, aliases of keys and ARNs of those aliases are supposed to work.
+    def test_all_types_of_key_id_can_be_used_for_encryption(
+        self, kms_client, kms_create_key, kms_create_alias
+    ):
+        def get_alias_arn_by_alias_name(kms_client, alias_name):
+            # list_aliases returns a paginated list, but we assume here that everything fits on one page
+            for alias_list_entry in kms_client.list_aliases(KeyId=key_id)["Aliases"]:
+                if alias_list_entry["AliasName"] == alias_name:
+                    return alias_list_entry["AliasArn"]
```

Comment:
While this is a fairly safe assumption here due to the fact that we're creating the alias via uid, we could also use a proper pagination here:

```suggestion
            for response in kms_client.create_pager("list_aliases").paginate(KeyId=key_id):
                for alias_list_entry in response["Aliases"]:
                    if alias_list_entry["AliasName"] == alias_name:
                        return alias_list_entry["AliasArn"]
```

Reply:
Done.
As you have told me outside of the PR, it is actually get_paginator() instead of create_pager().